lib/input.js

```diff
@@ -25,6 +25,9 @@ function _createInputDescriptor (input, inputOptions, containerOptions) {
     limitInputPixels: Math.pow(0x3FFF, 2),
     sequentialRead: false
   };
+  if (input && input.length === 0) {
+    throw Error('Input contains no data (its size is 0)');
```

Comment:
Please can you move this into the `if (is.buffer(input))` and `if (is.uint8Array(input)` blocks, then the error message can contain the type e.g. `Input Buffer is empty`.

Reply:
That's actually how I first implemented it, but then I changed it to this to avoid the redundancy. I checked the docs, and all accepted inputs have a `length` property, so it seemed appropriate to do as a catch-all. I considered the lack of type in the error message, but reasoned that it didn't provide anything particularly important since the problem is not related to the type (and the extra type information might lead to a red herring for the consumer).

Can change though.

I'm thinking to also update the test to include an assertion on the error's message to ensure it's the correct error (and not something later).